discovery/kubernetes/kubernetes.go

```diff
@@ -376,139 +395,64 @@ func (d *Discovery) Run(ctx context.Context, ch chan<- []*targetgroup.Group) {
 			go eps.podInf.Run(ctx.Done())
 		}
 	case RoleEndpoint:
-		for _, namespace := range namespaces {
-			e := d.client.CoreV1().Endpoints(namespace)
-			elw := &cache.ListWatch{
-				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-					options.FieldSelector = d.selectors.endpoints.field
```

Comment:
I see you're removing the custom ListWatch's here- you can also use the informerFactory's `InformerFor()` method to get an informer, that'll allow you to keep the custom ListWatch funcs

Reply:
Thanks. All of the Prometheus tests as well as my own manual testing are passing right now. Lets merge for  now and we can optimize for that in the future. That would probably make future merges easier if the code changes.